### PR TITLE
Onprem docs

### PR DESCRIPTION
**Pull request by J-Hunter-Hawke**
Added the `run-anywhere` directory in our website docs. I've intentionally left these files out of `contents.yaml` because we don't want them to show up on the page directly.

Please let me know if you think this will break things 😃

### CODE REVIEW

**Comment by tashian** [comment]
Hi! I'm reviewing this now.
I'm curious why, if we don't want these to be publicly visible, we don't just host them from Notion as locked pages?
Are we going allowing them to be indexed by search engines?

**Comment by tashian** [comment]
I'm getting parse errors on the pages when I try to look at them locally with the dev server. Some are because the pages are missing front matter, which is required. Eg:

```
---
title: Basic Certificate Authority Operations
html_title: basic certificate authority operations
description: Learn about the basic certificate authority operations of step-ca
---
```

File: src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+- [KOTS](https://kots.io/kots-cli/getting-started/#how-to-install)
+- [linkerd](https://linkerd.io/2.10/getting-started/#step-1-install-the-cli)
+- [step](https://smallstep.com/docs/step-cli/installation)
+
+### Terraform
+
+Create a GCP project and create infrastructure using instructions in the shared terraform repo.
+
+If you already have a Terraform environment set up for your account, add the [run-anywhere-terraform](https://github.com/smallstep/run-anywhere-terraform/tree/main/gcp) module to your configuration and follow the steps listed in its README, skipping the steps involving state creation. If you are new to Terraform, follow the [run-anywhere-terraform](https://github.com/smallstep/run-anywhere-terraform/tree/main/gcp) module’s README in its entirety: it will instruct you how to set up a Terraform [state](https://www.terraform.io/language/state) in your GCP account, instantiate the [module](https://www.terraform.io/language/modules/syntax), and run the subsequent Terraform [apply](https://www.terraform.io/cli/commands/apply).
```
Comment (tashian) [inline src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx]: If possible, I think we should move the caveats in this section into the Terraform repo README and just ask that they follow the instructions there.

File: src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+Alternatively, you may use the Terraform module as documentation and deploy the same set of resources following the policies required by your organization. While the Terraform module does offer most configurations to be specified by the user, it is by no means “one size fits all” - there are still some finite configuration details some organizations may need to tweak beyond the capability we allow.
+
+### DNS
+
+Refer to the GCP Console, Network Services ⇒ Cloud DNS ⇒ `default` zone to retrieve the list of nameservers for your GCP DNS configuration.
```
Comment (tashian) [inline src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx:72]: Could be nice to add: 

```
gcloud dns managed-zones describe default --flatten "nameServers[]" --format "value(nameServers)"
```

File: src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx (hunk 1 above)
Comment (tashian) [reply src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx]: On second thought, how about we inline all the terraform docs from that repo into this doc, so everything is in one place?

File: src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+```
+ns-cloud-c1.googledomains.com.
+ns-cloud-c2.googledomains.com.
+ns-cloud-c3.googledomains.com.
+ns-cloud-c4.googledomains.com.
+```
+
+## Databases
+
```
Comment (tashian) [inline src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx:91]: is this in Cloud SQL?

**Comment by tashian** [comment]
Hi @J-Hunter-Hawke overall this is a great starting point for these docs. Thanks for putting this together. I think a couple things are still needed:

- intro sentence or two on every page.
- front matter on every page
- lint & test on local dev server, make sure everything is rendering ok
- I added a couple comments — just a few smol things here & there

Once these are done I think we're good to go!

**Comment by J-Hunter-Hawke** [comment]
> Hi! I'm reviewing this now. I'm curious why, if we don't want these to be publicly visible, we don't just host them from Notion as locked pages? Are we going allowing them to be indexed by search engines?

@tashian  Reasoning here was to have "smallstep.com" links to hand out to customers who've signed contracts to use `run anywhere` but aren't cluttering up the docs pages for other groups. I honestly think this is still up in the air, though.

**Comment by J-Hunter-Hawke** [comment]
> I'm getting parse errors on the pages when I try to look at them locally with the dev server. Some are because the pages are missing front matter, which is required. Eg:
> 
> ```
> ---
> title: Basic Certificate Authority Operations
> html_title: basic certificate authority operations
> description: Learn about the basic certificate authority operations of step-ca
> ---
> ```

@tashian will add, thanks! TIL

File: src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx (hunk 1 above)
Comment (J-Hunter-Hawke) [reply src/pages/docs/run-anywhere/smallstep-platform-installation-gcp.mdx]: I'll pull those over once I have them finalized (probably later today / tomorrow morning)

**Comment by J-Hunter-Hawke** [comment]
- Added front matter
- Pulled in the Terraform READMEs
- Added intro paragraphs
- Adjusted some wording
- Linted and tested locally

**Pull request merged**